lib/rouge/lexers/cobol.rb

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class COBOL < RegexLexer
+      title 'COBOL'
+      desc 'COBOL (Common Business-Oriented Language) programming language'
+      tag 'cobol'
+      filenames '*.cob', '*.cbl'
+      mimetypes 'text/x-cobol'
+
+      # List of COBOL keywords
+      # sourced from https://www.ibm.com/docs/en/cobol-zos/6.4?topic=appendixes-reserved-words
+      KEYWORDS = %w[
```

Comment:
**suggestion:** Could we convert this into a class variable to take advantage of lazy loading? Constants are eagerly loaded when the class is defined, which can slow down startup time. Also, since the collection is rather big, Sets can be more memory-efficient due to their hash-based implementation.

```ruby
def keywords
  @keywords ||= Set.new(%w(ACCEPT ACCESS ...))
end

Reply:
I applied this, and two of your other suggestions. Hope it is what you meant, Ruby is a relatively new language to me so I had to do some learning about class variables and lazy loading.